src/util/structured_data.h

```diff
@@ -0,0 +1,99 @@
+// Copyright 2016-2020 Diffblue Limited. All Rights Reserved.
+#ifndef CPROVER_UTIL_STRUCTURED_DATA_H
+#define CPROVER_UTIL_STRUCTURED_DATA_H
+
+#include "json.h"
+#include "xml.h"
+#include <string>
+#include <vector>
+
+struct labelt
+{
+public:
+  explicit labelt(std::vector<std::string> components);
+
+  std::string camel_case() const;
+  std::string snake_case() const;
+  std::string kebab_case() const;
+  std::string pretty() const;
+
+  bool operator<(const labelt &other) const;
+
+private:
+  std::vector<std::string> components;
+};
+
+struct structured_data_entryt
+{
+  static structured_data_entryt data_node(const jsont &data);
+  static structured_data_entryt
+  entry(std::map<labelt, structured_data_entryt> children);
```

Comment:
I think `structured_data_entryt` unnecessarily exposes the fact that `jsont` and `std::map` are used to store the data and children. Could we instead have constructors which take string and int, and a method to add children? :bee:

Reply:
Don't agree with this - this allows you to easily create a constant structured_data using the existing type initialization C++ provides. Further, I'd either need to recreate constructors for all the json types, or limit what kind of json data you can output.

Reply:
Third option: a very small handful of specialized constructors/helper functions for the "common" cases, but leaving the full exposure of jsont/std::map for any of the less common cases?